packages/analysis-engine/src/parser.ts

```diff
@@ -1,121 +1,74 @@
-              return tagsMatrix[commitIdx].push(branchAndTagInfo.replace("tag: ", ""));
-            return branchesMatrix[commitIdx].push(branchAndTagInfo);
-          });
-        }
-        return false;
-      }
-      if (str.startsWith("Author:")) return getNameAndEmail(authors, str);
-      if (str.startsWith("AuthorDate")) return authorDates.push(new Date(str.split(": ")[1].trim()));
-      if (str.startsWith("Commit:")) return getNameAndEmail(committers, str);
-      if (str.startsWith("CommitDate")) {
-        let indexCheckFileChanged = idx + 2;
-        let eachCommitMessage = "";
-        while (splitByNewLine[indexCheckFileChanged] !== "") {
-          if (eachCommitMessage !== "") {
-            eachCommitMessage += "\n";
-          }
-          eachCommitMessage += splitByNewLine[indexCheckFileChanged].trim();
-          indexCheckFileChanged += 1;
+    // step 1: Extract commitData from the first line of the commit
+    const commitData = commitLines[0].split(GIT_LOG_SEPARATOR);
+    // Extract branch and tag data from commitData[2]
+    const refs = commitData[2].replace(" -> ", ", ").split(", ");
+    const [branches, tags]: string[][] = refs.reduce(
+      ([branches, tags], ref) => {
+        if (ref.startsWith("tag: ")) {
+          tags.push(ref.replace("tag: ", ""));
+        } else {
+          branches.push(ref);
         }
-        commitDates.push(new Date(str.split(": ")[1].trim()));
-        messages.push(eachCommitMessage);
-        commitTypes.push(getCommitMessageType(eachCommitMessage));
-      }
-      if (/^\d/.test(str) || /^-/.test(str)) {
-        const [addition, deletion, path] = str
-          .split(" ")
-          .filter((e) => e)[0]
-          .split("\t");
+        return [branches, tags];
+      },
+      [new Array<string>(), new Array<string>()]
+    );
 
-        const numberedAddition = addition === "-" ? 0 : Number(addition);
-        const numberedDeletion = deletion === "-" ? 0 : Number(deletion);
-        differenceStatistics[commitIdx].totalInsertionCount += numberedAddition;
-        differenceStatistics[commitIdx].totalDeletionCount += numberedDeletion;
-        differenceStatistics[commitIdx].fileDictionary[path] = {
-          insertionCount: numberedAddition,
-          deletionCount: numberedDeletion,
-        };
-      }
-      return false;
-    });
-  }
-
-  // 각 카테고리로 담은 다음 다시 JSON으로 변환하기 위함
-  const commitRaws: CommitRaw[] = [];
+    // make base commitRaw object
+    const commitRaw: CommitRaw = {
+      sequence: commitIdx - 1,
+      id: commitData[0],
+      parents: commitData[1].split(" "),
+      branches, // commitData[2] is already split into branches and tags
+      tags,
+      author: {
+        name: commitData[3],
+        email: commitData[4],
+      },
+      authorDate: new Date(commitData[5]),
+      committer: {
+        name: commitData[6],
+        email: commitData[7],
+      },
+      committerDate: new Date(commitData[8]),
+      message: commitData[9],
```

Comment:
여기 취향 차이긴 하지만 array destructuring으로 표시도 가능하겠네요!

```
const [id, parentsText, name, email, ......] = commitLines[0].split(GIT_LOG_SEPARATOR);
```

```suggestion
      id,
      parents: parentsText.split(" "),
      branches, // commitData[2] is already split into branches and tags
      tags,
      author: {
        name,
        email,
      },
      ...
```